src/audio/dai.c

```diff
@@ -66,6 +93,55 @@ struct dai_data {
 	uint64_t wallclock;	/* wall clock at stream start */
 };
 
+static void dai_atomic_trigger(void *arg, enum notify_id type, void *data);
+
+static int dai_add_group(struct comp_dev *dev, uint8_t group_id)
+{
+	struct list_item *dai_groups = &groups[cpu_get_id()].list;
+	struct dai_data *dd = comp_get_drvdata(dev);
+	struct list_item *group_item;
+	struct dai_group *group;
+
+	if (dd->group) {
+		if (dd->group->group_id != group_id) {
+			comp_err(dev, "dai_add_group(), DAI already in group");
+			return -EINVAL;
+		}
+
+		/* No need to re-assign to the same group, do nothing */
+		return 0;
+	}
+
+	list_for_item(group_item, dai_groups) {
```

Comment:
what are we looking for here.

Reply:
We're checking if a group of a given ID already exists.

Reply:
ok, so it's best to comment this for readability, I would recommend this for all list_for_each searches now.